src/containers/permissions/assign-permissions-to-object/StepSelectRoleOrUser.js

```diff
@@ -0,0 +1,145 @@
+  .some((value) => value && value.toLowerCase().includes(searchQuery.toLowerCase()));
+
+const StepSelectRoleOrUser = ({
+  existingPermissions,
+  organizationId,
+  setRoleOrUserPrincipleId,
+  setTargetRoleOrUserPrincipleType,
+  setTargetRoleOrUserTitle,
+  targetRoleOrUserPrincipal
+} :{
+  existingPermissions :Map<Principal, Map<List<UUID>, Ace>>;
+  organizationId :UUID;
+  setRoleOrUserPrincipleId :(id :string) => void;
+  setTargetRoleOrUserTitle :(title :string) => void;
+  setTargetRoleOrUserPrincipleType :(type :string) => void;
+  targetRoleOrUserPrincipal :?Principal;
+}) => {
+  const thisUserInfo = AuthUtils.getUserInfo() || { id: '' };
+  const thisUserId = thisUserInfo.id;
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const setQuery = (event :SyntheticInputEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+    setSearchQuery(value);
+  };
+
+  const organization :?Organization = useSelector(selectOrganization(organizationId));
+  const orgMembers :List<Map> = useSelector(selectOrganizationMembers(organizationId));
+
+  const userRoleOptions = useMemo(() => {
+    const options = [];
+    const permissionsExistOnRole = (optionPrincipal :?Principal) => existingPermissions
+      .some(({ principal } :Ace) => optionPrincipal && principalMatches(principal, optionPrincipal));
+    // add org members that don't already have permissions on object to options
+    orgMembers.toJS().forEach((member :Object) => {
+      const label = getUserTitle(member, thisUserId);
+      const memberId :string = getIn(member, ['profile', 'user_id']);
+      const principalValue :?Principal = getPrincipal(member);
+      if (!permissionsExistOnRole(principalValue) && stringsMatchQuery(searchQuery, [memberId, label])) {
+        options.push({
+          label,
+          principal: principalValue
+        });
+      }
+    });
+    // add org roles that don't already have permissions on object to options
+    organization?.roles.forEach((role :Object) => {
+      const label = role.title;
+      const roleId :string = role.id || '';
+      const principalValue :Object = role.principal || {};
+      if (!permissionsExistOnRole(principalValue) && stringsMatchQuery(searchQuery, [roleId, label])) {
+        options.push({
+          label,
+          principal: principalValue
+        });
+      }
+    });
+    return options;
+  }, [existingPermissions, organization, orgMembers, thisUserId, searchQuery]);
+
+  const handleOnChangeSelectRoleObject = (event :SyntheticInputEvent<HTMLInputElement>) => {
+    const { id, value, title } = event.currentTarget;
+    setRoleOrUserPrincipleId(id);
+    setTargetRoleOrUserPrincipleType(value);
+    setTargetRoleOrUserTitle(title);
+  };
+
+  return (
+    <StackGrid gap={32}>
+      <StackGrid>
+        <Typography>
+          Search for an existing user or role to create a permission for.
+        </Typography>
+        <SearchInput onChange={setQuery} searchQuery={searchQuery} />
```

Comment:
`searchQuery` is not a `SearchInput` prop, and also results in an error:

<img width="1920" alt="Screen Shot 2021-01-14 at 8 05 21 PM" src="https://user-images.githubusercontent.com/440299/104680047-d64d9a00-56a3-11eb-9e17-8d51ec0f6adf.png">